.github/workflows/build-and-cache.yaml

```diff
@@ -28,16 +28,23 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, macos-13]
+        os: [ ubuntu-latest, macos-latest, macos-13 ]
         packages:
-          - [holochain, hc, hc-run-local-services, hc-sandbox, hcterm]
-          - [lair-keystore]
-          - [hc-launch]
-          - [hc-scaffold]
+          - [ holochain, hc, hc-run-local-services, hc-sandbox, hcterm ]
+          - [ lair-keystore ]
+          - [ hc-launch ]
+          - [ hc-scaffold ]
+        include:
+          - os: ubuntu-latest
+            template: holonix-default
       fail-fast: false
 
     runs-on: ${{ matrix.os }}
 
+    permissions:
+      contents: read
+      id-token: write
```

Comment:
What is this for?

Reply:
Following the guide, it gets rid of some of the warnings/errors https://github.com/DeterminateSystems/magic-nix-cache?tab=readme-ov-file#magic-nix-cache

Doesn't deal with rate-limiting because it's an auto-provided access token, rather than one issued to a user